File: leetcodesln/RemoveDuplicatesFromSortedArray.cs
namespace leetcodesln
{
    public class RemoveDuplicatesFromSortedArray
    {
        public int RemoveDuplicates(int[] nums)
        {
            if(nums.Length <= 1) return nums.Length == 1? 1: 0;

            var currentPtr = 1;
            var sortedPtr = 0;
            while(currentPtr < nums.Length)
            {
                if(nums[sortedPtr]== nums[currentPtr])
                {
                    currentPtr++;
                }
                else
                {
                    nums[sortedPtr+1] = nums[currentPtr];
                    sortedPtr++;
                    currentPtr++;
                }
            }
            return sortedPtr+1;
        }
    }
}

File: Go/questions/RemoveDuplicatesFromSortedArray.go
package questions

func RemoveDuplicates(nums []int) int {
	n := len(nums)
	if n == 0 {
		return 0
	} else if n == 1 {
		return 1
	}

	cur, sorted := 1, 0

	for cur < n {
		if nums[cur] == nums[sorted] {
			cur++
		} else {
			nums[sorted+1] = nums[cur]
			sorted++
			cur++
		}
	}
	return sorted + 1
}
